model/maker.rb

```diff
@@ -0,0 +1,41 @@
+require 'data_mapper'
+require 'bcrypt'
+require_relative 'peep'
+require 'dm-validations'
+
+class Maker
+  include DataMapper::Resource
+
+  attr_reader :password
+
+  property :id, Serial
+  property :name, String, required: true
+  property :username, String, required: true, unique: true
+  property :email, String, required: true, unique: true
+  property :password_digest, Text, required: true
+
+  validates_format_of :email, as: :email_address
+
+  has n, :peeps
+
+  # Here, we are over-riding what gets stored in our database when we create a new Maker in our
+  # controller, i.e. we use Maker.create(password: params[:password]) (using the below password
+  # method as our key, instead of Maker.create(password_digest: params[:password]). This results
```

Comment:
Metrics/LineLength: Line is too long. [96/90] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)